test/new.jl

```diff
@@ -16,6 +16,41 @@ unicode_uuid = UUID("4ec0a83e-493e-50e2-b9ac-8f72acf5a8f5")
 unregistered_uuid = UUID("dcb67f36-efa0-11e8-0cef-2fc465ed98ae")
 simple_package_uuid = UUID("fc6b7c0f-8a2f-4256-bbf4-8c72c30df5be")
 
+using Downloads
+testurl = "https://api.github.com/repos/JuliaData/Parsers.jl/tarball/50c9a9ed8c714945e01cd53a21007ed3865ed714"
+for i in 1:10
+    @testset "Debugging: Downloads.download. Rep $i" begin
+        dest, io = mktemp()
+        Downloads.download(testurl, dest; headers = Pair{String, String}[], progress = (total, now) -> nothing)
+    end
+end
+for i in 1:10
+    @testset "Debugging: PlatformEngines.download. Rep $i" begin
+        dest, io = mktemp()
+        Pkg.PlatformEngines.download(testurl, dest)
+    end
+end
+for i in 1:10
+    @testset "Debugging: PlatformEngines.download, isolated. Rep $i" begin
+        isolate() do
+            dest, io = mktemp()
+            Pkg.PlatformEngines.download(testurl, dest)
+        end
+    end
+end
+
+for i in 1:10
+    @testset "Debugging: downloads. Rep $i" begin
+        isolate() do
+            @testset "tarball downloads" begin
+                Pkg.add("JSON"; use_only_tarballs_for_downloads=true)
+                @test "JSON" in [pkg.name for (uuid, pkg) in Pkg.dependencies()]
+                Pkg.rm("JSON")
+            end
+        end
+    end
+end
+
```

Comment:
I'm stumped.. can anyone see what's different about the first 3 sets of test to the final set? 
The `Downloads.download` at the core has the same args, from what i can tell.. yet only the last set will error 403